internal/database/db.go

```diff
@@ -0,0 +1,159 @@
+package database
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type Task struct {
+	Id      string `json:"id,omitempty"`
+	Date    string `json:"date,omitempty"`
+	Title   string `json:"title,omitempty"`
+	Comment string `json:"comment,omitempty"`
+	Repeat  string `json:"repeat,omitempty"`
+}
+
+type TasksStore struct {
+	Db *sql.DB
+}
+
+// CreateDB создает таблицу scheduler.
+func CreateDB(db *sql.DB) {
+	_, _ = db.Exec(`CREATE TABLE scheduler (
+        id INTEGER PRIMARY KEY AUTOINCREMENT,
+        date INTEGER NOT NULL DEFAULT 0,
+        title TEXT NOT NULL DEFAULT "",
+        comment TEXT NOT NULL DEFAULT "",
+        repeat VARCHAR(128) NOT NULL DEFAULT "");
+		CREATE INDEX scheduler_date ON scheduler (date);
+		`)
+}
+
+// CheckDB проверяет наличие БД с указанным именем, если нет, то создастся.
+func CheckDB(fileName string) {
+	path, err := os.Getwd()
+	if err != nil {
+		log.Println(err.Error())
+		return
+	}
+	dbFile := filepath.Join(path, "scheduler.db")
+
+	var has bool
+	if _, err := os.Stat(dbFile); err == nil {
+		has = true
+	}
+
+	if !has {
+		log.Println("Creating a database ...")
+		file, err := os.Create(dbFile)
+		if err != nil {
+			log.Println(err.Error())
+			return
+		}
+		defer file.Close()
+	}
+}
+
+// Метод PostTask добавляет задачу, с указанными параметрами в таблицу scheduler.
+func (tasks *Task) PostTask(db *sql.DB) (string, error) {
+	row, err := db.Exec("INSERT INTO scheduler (date, title, comment, repeat) VALUES (:date, :title, :comment, :repeat)",
+		sql.Named("date", tasks.Date), sql.Named("title", tasks.Title),
+		sql.Named("comment", tasks.Comment), sql.Named("repeat", tasks.Repeat))
+	if err != nil {
+		return "", err
+	}
+
+	id, err := row.LastInsertId()
+	if err != nil {
+		return "", err
```

Comment:
Ошибку можно обогащать
```suggestion
		return "", fmt.Errorf("error while getting last id: %w", err)
```

Reply:
не крит